airbyte-integrations/connectors/source-blob/source_blob/filereader.py

```diff
@@ -0,0 +1,144 @@
+        """ TODO docstring """
+
+    @abstractmethod
+    def stream_records(self, file) -> Iterator:
+        """ TODO: docstring """
+
+    @staticmethod
+    def json_type_to_pyarrow_type(typ: str, reverse=False, logger: AirbyteLogger=AirbyteLogger()) -> str:
+        """Convert Airbyte Type to PyArrow types to (or the other way around if reverse=True)
+        TODO: Docstring
+        """
+        str_typ = str(typ)
+        # this is a map of airbyte types to pyarrow types. The first list element of the pyarrow types should be the one to use where required.
+        map = {
+            "boolean": ("bool_", "bool"),
+            "integer": ("int64", "int8", "int16", "int32", "uint8", "uint16", "uint32", "uint64"),
+            "number": ("float64", "float16", "float32", "decimal128", "decimal256", "halffloat", "float", "double"),
+            "string": ("large_string", "string"),
+            "object": ("large_string", ),
+            "array": ("large_string", ),
+            "null": ("large_string", )
+        }
+        if not reverse:
+            for json_type, pyarrow_types in map.items():
+                if str_typ.lower() == json_type:
+                    return getattr(pa, pyarrow_types[0]).__call__()  # better way might be necessary when we decide to handle more type complexity
+            logger.warn(f"JSON type '{str_typ}' is unknown, falling back to default conversion to large_string")
+            return pa.large_string()
+        else:
+            for json_type, pyarrow_types in map.items():
+                if any([str_typ.startswith(pa_type) for pa_type in pyarrow_types]):
+                    return json_type
+            logger.warn(f"PyArrow type '{str_typ}' is unknown, falling back to default conversion to string")
+            return "string"  # default type if unspecified in map
+
+    @staticmethod
+    def json_schema_to_pyarrow_schema(schema: Mapping[str, Any], reverse: bool = False) -> Mapping[str, Any]:
+        """ TODO docstring """
+        new_schema = {}
+
+        for column, json_type in schema.items():
+            new_schema[column] = FileReader.json_type_to_pyarrow_type(json_type, reverse=reverse)
+
+        return new_schema
+
+
+class FileReaderCsv(FileReader):
+    """  TODO Docstring """
+    pass
+
+    @property
+    def is_binary(self):
+        return True
+
+    def _read_options(self):
+        return pa.csv.ReadOptions(
+            block_size=10000,
+            encoding=self._format.get("encoding", 'utf8')
+        )
+
+    def _parse_options(self):
+        return pa.csv.ParseOptions(
+            delimiter=self._format.get("delimiter", ','),
+            quote_char=self._format.get("quote_char", '"'),
+            double_quote=self._format.get("double_quote", True),
+            escape_char=self._format.get("escape_char", False),
+            newlines_in_values=self._format.get("newlines_in_values", False)
+        )
+
+    def _convert_options(self, json_schema=None):
```

Comment:
can you use typehints where possible in param names and return values?

Reply:
yeah absolutely, will do a 2nd pass through all the code type hinting and doc-stringing